archive_repo.py

```diff
@@ -683,34 +663,134 @@ def log(*str, **kwargs):
                 )
 
             # Collapse some nodes
-            collapse_to_login(pr, "author")
-            collapse_to_login(pr, "mergedBy")
+            collapse_map(pr, "author", "login")
+            collapse_map(pr, "mergedBy", "login")
             collapse_single(pr, "labels", "name")
+            collapse_map(pr, "baseRepository", "nameWithOwner")
+            collapse_map(pr, "headRepository", "nameWithOwner")
             collapse_single(pr, "assignees", "login")
             collapse(pr, "comments")
-            for comment in pr["comments"]:
-                collapse_to_login(comment, "author")
+            for comment in pr.get("comments", []):
+                collapse_map(comment, "author", "login")
             collapse(pr, "reviews")
-            for review in pr["reviews"]:
-                collapse_to_login(review, "author")
+            for review in pr.get("reviews", []):
+                collapse_map(review, "author", "login")
                 collapse(review, "comments")
 
             # Delete the old instance; add this instance
-            if number in pr_ref:
+            if not updateOld and number in pr_ref.keys():
                 del pr_ref[number]
-            pr_ref[number] = pr
+
+            if number in pr_ref.keys():
+                pr_ref[number].update(pr)
+            else:
+                pr_ref[number] = pr
             just_copy_old_file = False
 
         get_more_issues = prs["pageInfo"]["hasNextPage"]
         issue_cursor = prs["pageInfo"]["endCursor"]
 
         # Stop paginating if we've caught up to the last download
-        if prs["nodes"] and lastSuccess and not ref_is_issues_only:
+        if not updateOld and prs["nodes"] and lastSuccess:
             oldestRetrieved = dp.parse(prs["nodes"][-1]["updatedAt"])
             if oldestRetrieved < lastSuccess:
                 get_more_issues = False
 
-# Fetch the Labels
+
+#########################
+## Upgrade definitions ##
+#########################
+
+current_magic = "E!vIA5L86J2I"
+
+upgrades = {
+    "B8n2c@e8kvfx": {
+        "result": "E!vIA5L86J2I",
+        "issues": None,
+        "prs": """
+fragment prFields on PullRequest {
+    number
+    baseRepository { nameWithOwner }
+    baseRefName
+    baseRefOid
+    headRepository { nameWithOwner }
+    headRefName
+    headRefOid
+    mergeCommit { oid }
+}
+""",
+    }
+}
+
+#####################
+## Body of program ##
+#####################
+
+just_copy_old_file = False
+(owner, repo) = args.repo.split("/", 1)
+
+## Read in the reference files, if any
+issue_ref = dict()
+pr_ref = dict()
+lastSuccess = None
+ref_is_issues_only = True
+
+if args.refFile:
+    try:
```

Comment:
It seems like this might be worth pushing into a new function.  Or two, called `load_reference` and `update_reference`.  The first can do the simple loading, ignorant of any upgrade logic.  The second can turn the loaded reference into a reference of the current version.